src/LiveSplit.Core/Model/RunMetadata.cs

```diff
@@ -316,4 +349,26 @@ private void TriggerPropertyChanged(bool clearRunID)
             PropertyChanged?.Invoke(this, new MetadataChangedEventArgs(clearRunID));
         }
     }
+
+    internal class CustomVariable
+    {
+        public string Value { get; set; } = null;
+        // monotonic: once true, never set back to false
+        public bool IsPermanent { get; private set; } = false;
```

Comment:
Assignment of default value is unnecessary.

```suggestion
        public string Value { get; set; }
        // monotonic: once true, never set back to false
        public bool IsPermanent { get; private set; }
```